Python/tigre/Source/_AwminTV.pyx

```diff
@@ -31,6 +31,7 @@ def cuda_raise_errors(error_code):
         raise ValueError('TIGRE:Call to aw_pocs_tv failed')
 
 def AwminTV(np.ndarray[np.float32_t, ndim=3] src,float alpha = 15.0,int maxiter = 100, float delta=-0.005):
+    src=np.transpose(src,(1,2,0)).copy()  # shift 1st dim to last, match MATLAB order

```

Comment:
I agree that this makes it the same as MATLAB, but I am suprised this makes a difference. 

In theory, the CUDA code here does not care about what is X,Y,Z axis, so my assumption would be that this line should do nothing, the output should be the same with and without it. 

If this is not the case, then I am a bit confused....

Reply:
@yliu88au Great work! Two questions:
1. Could you please try anisotropic case like
```
geo.nVoxel = np.array([1, 256, 512])
```
to see which of 
```
src=np.transpose(src,(1,2,0)).copy()
```
and 
```
src=np.transpose(src,(2,1,0)).copy()
```
is correct?

2. Did you try swapping like as follows
```
    cdef long imgsize[3]
    imgsize[0] = <long> size_img[2]
    imgsize[1] = <long> size_img[1]
    imgsize[2] = <long> size_img[0]
```
instead of ```transpose``` and ```copy```?

I'm sorry I cannot test by myself now.

Reply:
I think @genusn second solution is what can make MATLAB and python do the same without the `np.transpose` call. 
Unfortunately I have not managed to get my python working yet so I can not test it yet! Its also holiday season where I am from so I will be a bit more sparse these days. But there is no hurry, we can fix this in few weeks otherwise.

Reply:
This is is very interesting. I have tested @genusn second solution and it works for my particular 2D example. However, if I use the following swapping or without swapping, it won't work.


    imgsize[0] = <long> size_img[2]

    imgsize[1] = <long> size_img[0]

    imgsize[2] = <long> size_img[1]

The difference between with or without swapping is very significant. The POCS stop criteria "c" changes from
c=-0.21086 (without) to c=-0.9309 (with swapping). MATLAB version c=-0.96..., The remaining difference may be due to FDK initialization difference between Python and MATLAB version.

Reply:
Further test using the same initial image (all zeros) shows that both MATLAB and Python versions yield very similar "c" values for the OS_AwASD_POCS algorithm (MATLAB c=-0.821. Python c=-0.831) for my 2D real data example. Happy with this result.

Reply:
Continuing test shows that the following two types of working swapping in Python gave identical results for "square" case, ie., nVoxel=[h,n,n], where h can be 1 (2D case) or h>1 (3D case):
swapping 1: @genusn 2.
 
swapping 2:

    imgsize[0] = <long> size_img[1]
    imgsize[1] = <long> size_img[2]
    imgsize[2] = <long> size_img[0]

For non-squared case (for example, 128x256x512), the results are slightly different. It is a bit hard for me to judge which one is correct. However, it seems @genusn 's swapping is slightly better for both 2D and 3D cases.

Reply:
I am almost certain that the correct one is:

```
imgsize[0] = <long> size_img[2]
imgsize[1] = <long> size_img[1]
imgsize[2] = <long> size_img[0]
```

Did this not work?

Reply:
This does work, and it is the better one.

Reply:
How about here? 
https://github.com/CERN/TIGRE/blob/41ffa73421512b49515e749a7d809d8d8c0af948/Python/tigre/Source/_tvdenoising.pyx#L53-L55

Reply:
@genusn Good point, I will look at it and do some tests.

Reply:
@genusn I have made the same change as in others to _tvdenoising.pyx:

    imgsize[0] = <long> size_img[2]
    imgsize[1] = <long> size_img[1]
    imgsize[2] = <long> size_img[0]

The test results seems make no difference here (still has small difference between Python and MATLAB, due to the different FDK init  images? #229 ). But for sake of consistency, we should make this change as well.